lib/modules/manager/npm/extract/pnpm.ts

```diff
@@ -135,3 +137,123 @@ export async function detectPnpmWorkspaces(
     }
   }
 }
+
+export async function getPnpmLock(filePath: string): Promise<LockFile> {
+  // TODO #7154
+  const pnpmLockRaw = (await readLocalFile(filePath, 'utf8'))!;
+  try {
+    const lockParsed = load(pnpmLockRaw) as Record<string, any>;
+    if (!lockParsed) {
```

Comment:
```suggestion
    if (!is.plainObject(lockParsed)) {
```

Reply:
Updated suggestions

Reply:
Won't it be ok to not check it since we have a catch statement that can handle invalid parsed content.?